examples/D3D12Binding.cpp

```diff
@@ -103,40 +103,60 @@ class D3D12Binding : public BackendBinding {
 
             rtvDescriptorSize = d3d12Device->GetDescriptorHandleIncrementSize(D3D12_DESCRIPTOR_HEAP_TYPE_RTV);
 
-            // Create a RTV for each frame.
+            // Create a RTV and command allocators for each frame.
             D3D12_CPU_DESCRIPTOR_HANDLE renderTargetViewHandle = renderTargetViewHeap->GetCPUDescriptorHandleForHeapStart();
             for (uint32_t n = 0; n < kFrameCount; ++n) {
                 ASSERT_SUCCESS(swapChain->GetBuffer(n, IID_PPV_ARGS(&renderTargetResources[n])));
                 d3d12Device->CreateRenderTargetView(renderTargetResources[n].Get(), nullptr, renderTargetViewHandle);
                 renderTargetViewHandle.ptr += rtvDescriptorSize;
+
+                ASSERT_SUCCESS(d3d12Device->CreateCommandAllocator(D3D12_COMMAND_LIST_TYPE_DIRECT, IID_PPV_ARGS(&commandAllocators[n])));
             }
 
-            ASSERT_SUCCESS(d3d12Device->CreateFence(fenceValues[frameIndex], D3D12_FENCE_FLAG_NONE, IID_PPV_ARGS(&fence)));
-            fenceValues[frameIndex]++;
+            ASSERT_SUCCESS(d3d12Device->CreateCommandList(
+                0,
+                D3D12_COMMAND_LIST_TYPE_DIRECT,
+                commandAllocators[frameIndex].Get(),
+                nullptr,
+                IID_PPV_ARGS(&commandList)
+            ));
+
+            currentFrameNumber = 0;
+            ASSERT_SUCCESS(d3d12Device->CreateFence(currentFrameNumber, D3D12_FENCE_FLAG_NONE, IID_PPV_ARGS(&fence)));
+            fenceValues[frameIndex] = ++currentFrameNumber;
             fenceEvent = CreateEvent(nullptr, FALSE, FALSE, nullptr);
             ASSERT(fenceEvent != nullptr);
 
-            SetNextRenderTarget();
+            SetNextRenderTarget(frameIndex);
         }
         void SwapBuffers() override {
-            // Present the rendered frame
+            // Execute commands to present the current frame's render target
+            PresentFrame(frameIndex);
```

Comment:
PresentFrame is only called here, either it could be inlined here, or the swapChain->Present call should be moved inside PresentFrame (and PresentFrame renamed to something like EnqueuePresentCommands).